### PR TITLE
Improve helix failure resilience with API call retries

### PR DESCRIPTION
**Pull request by chingucoding**
## Description
Improve CIs resilience against API call failures by retrying API calls and downloads on failure. See https://github.com/microsoft/microsoft-ui-xaml/pull/4108#issuecomment-782341026 for more information.

@kmahone FYI

## Motivation and Context

No associated issue.
## How Has This Been Tested?

Not tested :(

### CODE REVIEW

**Review by kmahone — DISMISSED** [review]

**Comment by kmahone** [comment]
Looks good. Thanks for making this change.

File: build/Helix/UpdateUnreliableTests.ps1 [hunk 1]
```diff
@@ -54,7 +54,7 @@ foreach ($testRun in $testRuns.value)
             Write-Host "  Test $($testResult.testCaseTitle) was detected as unreliable. Updating..."
             
             # The errorMessage field contains a link to the JSON-encoded rerun result data.
-            $rerunResults = ConvertFrom-Json (New-Object System.Net.WebClient).DownloadString($testResult.errorMessage)
+            $rerunResults = ConvertFrom-Json Download-StringWithRetries $testResult.errorMessage
```
Comment (kmahone) [inline build/Helix/UpdateUnreliableTests.ps1:57]: This function Download-StringWithRetries takes two arguments. Either update the helper function to only take one argument, or pass a 'file name' here. Also, I am not sure now PS handles precedence, so you might want to add parens. Or do something like:

$resultsJson = Download-StringWithRetries $testResult.errorMessage
$rerunResults = ConvertFrom-Json $resultsJson

**Review by kmahone — CHANGES_REQUESTED** [review]
:clock1:

File: build/Helix/UpdateUnreliableTests.ps1 (hunk 1 above)
Comment (chingucoding) [reply build/Helix/UpdateUnreliableTests.ps1]: Good point, addressed both issues now. Going with keeping up the file name to make later WinUI 3 merges easier and help debugging issues down the line.

**Comment by kmahone** [comment]
I ran the script locally and it works now. So we should be good on the next Pipeline run.

**Review by kmahone — APPROVED** [review]

**Pull request merged**